pdata/internal/generated_pmetric.go

```diff
@@ -1444,24 +1444,24 @@ func (ms HistogramDataPoint) SetSum(v float64) {
 	(*ms.orig).Sum_ = &otlpmetrics.HistogramDataPoint_Sum{Sum: v}
 }
 
-// MBucketCounts returns the mbucketcounts associated with this HistogramDataPoint.
-func (ms HistogramDataPoint) MBucketCounts() []uint64 {
-	return (*ms.orig).BucketCounts
+// BucketCounts returns the bucketcounts associated with this HistogramDataPoint.
+func (ms HistogramDataPoint) BucketCounts() ImmutableUInt64Slice {
+	return ImmutableUInt64Slice{(*ms.orig).BucketCounts}
 }
 
-// SetMBucketCounts replaces the mbucketcounts associated with this HistogramDataPoint.
-func (ms HistogramDataPoint) SetMBucketCounts(v []uint64) {
-	(*ms.orig).BucketCounts = v
+// SetBucketCounts replaces the bucketcounts associated with this HistogramDataPoint.
+func (ms HistogramDataPoint) SetBucketCounts(v ImmutableUInt64Slice) {
+	(*ms.orig).BucketCounts = v.value
 }
 
-// MExplicitBounds returns the mexplicitbounds associated with this HistogramDataPoint.
-func (ms HistogramDataPoint) MExplicitBounds() []float64 {
-	return (*ms.orig).ExplicitBounds
+// ExplicitBounds returns the explicitbounds associated with this HistogramDataPoint.
+func (ms HistogramDataPoint) ExplicitBounds() ImmutableFloat64Slice {
+	return ImmutableFloat64Slice{(*ms.orig).ExplicitBounds}
```

Comment:
nit: use named arguments for initialization?

Reply:
Updated